caffe2/operators/conv_gradient_op.cc

```diff
@@ -16,35 +16,71 @@ class GetConvGradient : public GradientMakerBase {
 
     auto compute_dX = !argsHelper.GetSingleArgument<bool>("no_gradient_to_input", 0);
 
-    if (def_.input_size() == 3) {
+    auto engine = def_.engine();
+
+    // Use seperate gradient ops
+    if (engine == "CUDNN") {
```

Comment:
There is still some conv tests failing because of running host with engine='CUDA'.
We can fix this by checking that device is CUDA:: 
`if (engine == "CUDNN" && def._ device_option().device_type == 1)`